code/modules/vehicles/mecha/combat/savannah_ivanov.dm

```diff
@@ -0,0 +1,277 @@
+	animate(src, pixel_z = 400, time = 10, easing = QUAD_EASING|EASE_IN, flags = ANIMATION_PARALLEL) //Animate our rising mech (just like pods hehe)
+	addtimer(CALLBACK(src, .proc/begin_landing, pilot), 2 SECONDS)
+
+/obj/vehicle/sealed/mecha/combat/savannah_ivanov/proc/shake_for(duration, amt)
+	var/offset = prob(50) ? -amt : amt
+	var/old_pixel_x = pixel_x
+	animate(src, pixel_x = pixel_x + offset, time = 0.2, loop = -1, flags = ANIMATION_PARALLEL) //start shaking
+	addtimer(CALLBACK(src, .proc/stop_shaking, old_pixel_x), duration)
+
+/obj/vehicle/sealed/mecha/combat/savannah_ivanov/proc/stop_shaking(old_px)
+	if(skyfall_charge_level == SKYFALL_CHARGELEVEL_LAUNCH)
+		animate(src)
+	pixel_x = old_px
+
+/obj/vehicle/sealed/mecha/combat/savannah_ivanov/proc/begin_landing(mob/living/pilot)
+	animate(src, pixel_z = 0, time = 10, easing = QUAD_EASING|EASE_IN, flags = ANIMATION_PARALLEL)
+	animate(src, alpha = 255, time = 8, easing = QUAD_EASING|EASE_IN, flags = ANIMATION_PARALLEL)
+	addtimer(CALLBACK(src, .proc/land, pilot), 10)
+
+/obj/vehicle/sealed/mecha/combat/savannah_ivanov/proc/land(mob/living/pilot)
+	visible_message("<span class='danger'>[src] crashes into the ground!</span>")
+	playsound(src, 'sound/effects/explosion1.ogg', 50, 1)
+	resistance_flags &= ~INDESTRUCTIBLE
+	mecha_flags &= ~(QUIET_STEPS|QUIET_TURNS|CANNOT_INTERACT)
+	phasing = initial(phasing)
+	movedelay = initial(movedelay)
+	density = TRUE
+	layer = initial(layer)
+	skyfall_charge_level = 0
+	update_icon_state()
+	for(var/turf/open/floor/tiled_turf in range(1, src))
+		tiled_turf.break_tile()
+	var/turf/landed_on = get_turf(src)
+	if(isclosedturf(landed_on))
+		landed_on.ScrapeAway()
+	for(var/mob/living/victim in landed_on)
+		to_chat(victim, "<span class='userdanger'>[src] lands on you from above!</span>")
+		if(victim.stat != CONSCIOUS)
+			victim.gib(FALSE, FALSE, FALSE)
+		else
+			victim.adjustBruteLoss(80)
+
+/obj/vehicle/sealed/mecha/combat/savannah_ivanov/proc/abort_skyfall(mob/pilot)
+	if(pilot)
+		to_chat(pilot, "[icon2html(src, pilot)]<span class='notice'>Skyfall aborted.</span>")
+	COOLDOWN_START(src, skyfall_cooldown, skyfall_charge_level * 10 SECONDS) //so aborting skyfall later in the process imposes a longer cooldown
+	skyfall_charge_level = 0
+
+/obj/vehicle/sealed/mecha/combat/savannah_ivanov/proc/start_missile_targeting(mob/gunner, silent = TRUE)
+	if(!silent)
+		to_chat(gunner, "<span class='warning'>Ivanov Strike targeting process booted.</br>\
+		Your next click will fire the missile (provided the mech is facing the right direction).</span>")
+	aiming_ivanov = TRUE
+	RegisterSignal(src, COMSIG_MECHA_MELEE_CLICK, .proc/on_melee_click)
+	RegisterSignal(src, COMSIG_MECHA_EQUIPMENT_CLICK, .proc/on_equipment_click)
+	gunner.client.mouse_override_icon = 'icons/effects/mouse_pointers/supplypod_down_target.dmi'
+	gunner.update_mouse_pointer()
+	gunner.overlay_fullscreen("ivanov", /atom/movable/screen/fullscreen/ivanov_display, 1)
```

Comment:
Overlay won't fully cover screen for those with widescreen preference toggled

Reply:
this is... either an issue with all fullscreens, or none of them. either way, not MY problem >:]

Reply:
I checked, it ends up covering widescreen! 🎉